java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/docker/StageDockerArtifactBuildStep.java

```diff
@@ -1,151 +0,0 @@
-/*
- * Copyright 2017 Google Inc. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *       http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.google.cloud.runtimes.builder.buildsteps.docker;
-
-import com.google.cloud.runtimes.builder.buildsteps.base.BuildStep;
-import com.google.cloud.runtimes.builder.buildsteps.base.BuildStepException;
-import com.google.cloud.runtimes.builder.buildsteps.base.BuildStepMetadataConstants;
-import com.google.cloud.runtimes.builder.config.domain.RuntimeConfig;
-import com.google.cloud.runtimes.builder.exception.ArtifactNotFoundException;
-import com.google.cloud.runtimes.builder.exception.TooManyArtifactsException;
-import com.google.inject.Inject;
-
-import com.google.inject.assistedinject.Assisted;
-
-import org.apache.commons.io.FileUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import java.io.BufferedWriter;
-import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
-
-public class StageDockerArtifactBuildStep extends BuildStep {
-
-  private static final String DOCKER_STAGING_DIR = ".docker_staging";
```

Comment:
Since we're just generating a `Dockerfile` and `.dockerignore`, I think it makes more sense to put them directly in the source directory, since staging isn't really necessary anymore - all build actions can happen in the source tree itself. IMO reintroducing staging would add more unnecessary complexity.

I hadn't really been focusing on the local use case - is that something you think should be considered more heavily?